src/handlers/retryHandler.ts

```diff
@@ -109,10 +109,19 @@ export const retryRequest = async (
       }
     );
   } catch (error: any) {
-    lastResponse = new Response(error.message, {
-      status: error.status,
-      headers: error.headers,
-    });
+    if (error instanceof TypeError && error.message?.includes('fetch failed')) {
```

Comment:
This check should not be present in the retryHandler, the string is generic and this might get caught unintentionally

Reply:
You can throw this from inside `TritonErrorResponseTransform`

Reply:
@narengogi I get the point. The error message handling was generic here. 

The error comes in case the user has provided incorrect IP Address of `custom_host` while we are making the connection with the server. At that time, the code doesn't reach till `TritonErrorResponseTransform`.

I have improved the error handling by specifically handling the `ConnectionTimeOutError`. Please note that this error can come in case of `Ollama` connection as well.

Reply:
I think it should be fine if there's multiple retries in case the host can't be reached, that is expected, checking the error with a string should be strictly disallowed

Reply:
Yes @narengogi . Earlier, there were no retries in case of connection timeout. This change helps in getting expected functionality. 
Also, I have updated the code and made it specific to ConnectionTimeOutError. Please see this:

https://github.com/Portkey-AI/gateway/pull/445/files#diff-e070ec0094d3af60952ce2f32b6d12534a6d84a6a15e3f425cdf45766ef5477cR112